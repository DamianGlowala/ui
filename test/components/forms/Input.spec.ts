import { describe, it, expect } from 'vitest'
import { UInput } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Input', () => {
  it.each([
    ['basic case', {}],
    ['with custom name and placeholder', { props: { name: 'username', placeholder: 'Enter your username' } }],
    ['with value', { props: { value: 'JohnDoe' } }],
    ['with disabled', { props: { disabled: true } }],
    ['with readonly', { props: { readonly: true } }],
    ['with autofocus', { props: { autofocus: true } }],
    ['with required', { props: { required: true } }],
    ['with custom class', { props: { class: 'custom-class' } }],
    ['with custom style', { props: { style: { color: 'red', fontSize: '16px' } } }],
    ['with custom type', { props: { type: 'password' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Input.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UInput)
    expect(html).toMatchSnapshot()
  })
})
