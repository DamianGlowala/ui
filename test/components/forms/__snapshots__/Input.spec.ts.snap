// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > renders basic case correctly 1`] = `
"<div class=""><input type="text" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with autofocus correctly 1`] = `
"<div class=""><input type="text" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with custom class correctly 1`] = `
"<div class=""><input type="text" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with custom name and placeholder correctly 1`] = `
"<div class=""><input name="username" type="text" placeholder="Enter your username" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with custom style correctly 1`] = `
"<div class=""><input type="text" class="" style="color: red; font-size: 16px;">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with custom type correctly 1`] = `
"<div class=""><input type="password" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with disabled correctly 1`] = `
"<div class=""><input type="text" disabled="" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with readonly correctly 1`] = `
"<div class=""><input type="text" class="" readonly="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with required correctly 1`] = `
"<div class=""><input type="text" required="" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Input > renders with value correctly 1`] = `
"<div class=""><input type="text" class="">label
  <!--v-if-->
  <!--v-if-->
</div>"
`;
